scripts/invariant_checks.py

```diff
@@ -67,11 +72,14 @@ def main(argv: Sequence[str] | None = None) -> None:
     # Get deployed pools on first iteration
     deployed_pools = Hyperdrive.get_hyperdrive_pools_from_registry(chain, registry_address)
 
+    start_block_number = chain.block_data().get("number", None)
```

Comment:
Is there a reason you use `chain.block_data().get("number", None)` here and `chain.block_number()` below? I looked and they seem equivalent to me, maybe the web3 call under the hood for `chain.block_number()`, which uses `RPC.eth_blockNumber` ends up making two http requests? If there's no difference, I like `chain.block_number()` as the standard only because it's more readable.